### PR TITLE
Refs #3866. Fixed Serial read bug [4479]

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request fixes a bug in the Serial Protocol.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Linux/115/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Windows/75/)
* NuttX [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20NuttX/56/)

File: src/c/core/communication/serial_protocol.c [hunk 1]
```diff
@@ -257,7 +257,7 @@ size_t uxr_read_serial_msg(uxrSerialIO* serial_io,
         }
     }
 
-    if (0 < (bytes_read[0] + bytes_read[1]))
+    if (0 < (bytes_read[0] + bytes_read[1]) || (serial_io->rb_tail != serial_io->rb_head))
```
Comment (lemunozm) [inline src/c/core/communication/serial_protocol.c]: Does exist the case where the first part is false and the second is true? 
If no bytes were read, the head and the tail of the second part have always the same position?

Comment (julianbermudez) [reply src/c/core/communication/serial_protocol.c]: Yes, it occurs when you do not read anything from the hardware but you have data in the processing buffer due to the previous hardware reading, that is, when you read two message from a serial reading.

Comment (lemunozm) [reply src/c/core/communication/serial_protocol.c]: ok, great!

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Linux/119/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Windows/80/)
* NuttX [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20NuttX/61/)

**Pull request merged**